tests/zfs-tests/tests/functional/delegate/delegate_common.kshlib

```diff
@@ -53,15 +53,11 @@ function cleanup_user_group
 #
 function restore_root_datasets
 {
-	if datasetexists $ROOT_TESTFS ; then
-		log_must zfs destroy -Rf $ROOT_TESTFS
-	fi
+	destroy_dataset "$ROOT_TESTFS" "-Rf"
 	log_must zfs create $ROOT_TESTFS
 
 	if is_global_zone ; then
-		if datasetexists $ROOT_TESTVOL ; then
-			log_must zfs destroy -Rf $ROOT_TESTVOL
-		fi
+		destroy_dataset "$ROOT_TESTVOL" "-Rf"
```

Comment:
There appear to be a few other common places which could benefit from calling `destroy_dataset`.